hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -183,9 +184,9 @@ case class HoodieFileIndex(spark: SparkSession,
           }).filter(slice => slice != null)
             .map(fileInfo => new FileStatus(fileInfo.getLength, fileInfo.isDirectory, 0, fileInfo.getBlockSize,
               fileInfo.getModificationTime, new Path(fileInfo.getPath.toUri)))
-          val c = fileSlices.filter(f => (includeLogFiles && f.getLogFiles.findAny().isPresent)
-            || (f.getBaseFile.isPresent && f.getBaseFile.get().getBootstrapBaseFile.isPresent)).
-            foldLeft(Map[String, FileSlice]()) { (m, f) => m + (f.getFileId -> f) }
+          val c = fileSlices.filter(f => f.hasBootstrapBase || (includeLogFiles && f.hasLogFiles))
```

Comment:
After rethinking the changes, I think we should change the implementation of `filterFileSlices` instead of changing the returned file slices afterwards.  The logic here should include transformations for adapting to Spark implementation.

The reason is that, the data skipping happens in `filterFileSlices` and prunes the file slices returned by `prunePartitionsAndGetFileSlices`, which can contain the log files.  For read-optimized queries, the data skipping should depend on the file slices without log files; otherwise, the pruning can be potentially wrong (data skipping considers log files, while the file listing here does not).

Reply:
Also, the caller instantiating the file index needs to make sure the correct argument `includeLogFiles` is passed in.